### ISSUE TITLE
Configure runtime augument for java native image

### ISSUE THREAD

**Issue by zhoufenqin** [paketo-buildpacks/native-image#252]
I use kpack 0.9.2 and java-native-image.

From GraalVM [Debugging and Diagnostics](https://www.graalvm.org/22.3/reference-manual/native-image/debugging-and-diagnostics/) doc, I want to configure heapdump, see the doc: https://www.graalvm.org/22.3/reference-manual/native-image/guides/create-heap-dump/

In build phase, add `BP_NATIVE_IMAGE_BUILD_ARGUMENTS="--enable-monitoring=heapdump"`
In runtime phase, I hope to run `{command} -XX:+DumpHeapAndExit`, but there is no place to pass `-XX:+DumpHeapAndExit`. I directly run the command in a container, the error shows:

![image](https://user-images.githubusercontent.com/8765864/233843946-9a0b66b8-3412-447a-903e-a84393b5edb7.png)

## Current Behavior
should pass some argument to runtime phase

## Steps to Reproduce

1.

**Comment by dmikusa** [comment]
1. I think you want this -> https://buildpacks.io/docs/app-developer-guide/run-an-app/#default-process-type-with-additional-arguments. That will let you add arguments to the default process which should be your application.

2. You could include a Procfile & specify the command that way but it's not really necessary because of 1.).

I haven't tried it, but you might also try sending the SIGUSR1 signal. That might let you trigger a heap dump without modifying the start command.

**Comment by zhoufenqin** [comment]
> 1. I think you want this -> https://buildpacks.io/docs/app-developer-guide/run-an-app/#default-process-type-with-additional-arguments. That will let you add arguments to the default process which should be your application.
> 2. You could include a Procfile & specify the command that way but it's not really necessary because of 1.).
> 
> I haven't tried it, but you might also try sending the SIGUSR1 signal. That might let you trigger a heap dump without modifying the start command.

Hi @dmikusa set the default process with arguments using https://buildpacks.io/docs/app-developer-guide/run-an-app/#default-process-type-with-additional-arguments do not work well because Java native image layer is not put into the runtime image.
![image](https://user-images.githubusercontent.com/8765864/233903379-6e42ea93-4488-4175-b547-82c0bcf38398.png)

![image](https://user-images.githubusercontent.com/8765864/233903399-947b49b1-21cf-4e39-85a2-f0cef6cca7f8.png)

the /layers show only the ca-certificate buildpack layer is put into the final container image.
![image](https://user-images.githubusercontent.com/8765864/233903544-9c3a3f3c-f7c3-46d7-a462-8ecc059aa7d7.png)


So it may need the buildpack to do some changes to support the heapdump/jfr shoubleshooting, like the default path for the dump file needs to be changed.

Besides, I didn't use the `SIGUSR1 signal` because I don't want to send a signal from an application, hope to use a heap dump like in JVM way (e.g. set `BPL_HEAP_DUMP_PATH` when using JDK)

**Comment by dmikusa** [comment]
I'm not sure why they would remove the ability to set the heap dump path. Maybe ask the GraalVM team? We can't really do anything without that option because even if we published the native-image layer, that would be a read-only layer so it couldn't be written to. I'm sure we're not the only ones creating images where the JVM is on a read-only layer. It has to have some way to write the heap dump elsewhere.

You might also ask if they plan to add an option like `-XX:+HeapDumpOnOutOfMemoryError`. We'd need both of those to enable `BPL_HEAP_DUMP_PATH` to work the same for a native-image binary.

**Comment by zhoufenqin** [comment]
The default path for heap dump file is ` /layers/paketo-buildpacks_native-image/native-image`  rather than `/workspace`， I think it's because the `native-image` command has params like this:
![image](https://user-images.githubusercontent.com/8765864/234183301-3a760af1-1041-4508-bc79-9cbd23e36c11.png)

```
-H:Name=/layers/paketo-buildpacks_native-image/native-image/io.paketo.demo.DemoApplication
```

The executable file `io.paketo.demo.DemoApplication` is finally put into `/workspace/io.paketo.demo.DemoApplication`

Is it a bug in native-image buildpack? see https://github.com/paketo-buildpacks/native-image/blob/main/native/arguments.go#L159

**Comment by dmikusa** [comment]
No, that's intentional. The native-image file is put in the layer so that it can be cached. If you rebuild and your application hasn't changed, then it'll reuse that layer and skip the expensive build.

Could it be done differently? Yes. There are probably 5 different ways you could technically handle this, but presently it's working the way it was designed.
